gateway/modules/schema/update.go

```diff
@@ -164,6 +176,43 @@ func validateDateOperations(col string, doc interface{}, SchemaDoc model.Fields)
 	return nil
 }
 
+func (s *Schema) validateUnsetOperation(dbAlias, col string, doc interface{}, schemaDoc model.Fields) error {
+	v, ok := doc.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("document not of type object in collection %s", col)
+	}
+	dbType, err := s.crud.GetDBType(dbAlias)
+	if err != nil {
+		return err
+	}
+	if dbType == string(utils.Mongo) {
+		for fieldName := range v {
+			columnInfo, ok := schemaDoc[fieldName]
+			if ok {
+				if columnInfo.IsFieldTypeRequired {
+					return fmt.Errorf("cannot use $unset on field which is required")
+				}
+			}
+			// TODO CAN SPECIFIED FIELD NAME DOESN'T EXIST IN SCHEMA IN CASE OF MONGO
+		}
+		return nil
+	}
+
+	if dbType == string(utils.Postgres) || dbType == string(utils.MySQL) || dbType == string(utils.SQLServer) {
+		for fieldName := range v {
+			columnInfo, ok := schemaDoc[fieldName]
```

Comment:
Fieldname is in the form of `f1.f2.f3`. only the top level field will be in the schema